### ISSUE TITLE
Eliminate OVB from the valid Chapman Codes

### ISSUE THREAD

**Issue by geoffj-FUG** [FreeUKGen/FreeCENMigration#1310]
Following recent email discussions I have created this Action to eliminate OVB from our Chapman Codes.
FreeCEN has two Chapman Codes for Overseas – OVB and OVF. Only one of these is a genuine Chapman Code – OVF. See one of the lists of World Chapman Codes at http://home.vicnet.net.au/~mgfhs/chworld.htm or any other site that lists the recognised Chapman Codes. 
OVB was an invention of FreeCEN right at the beginning. It appears to have been an attempt to capture the British Subjects because the spreadsheet had not allowed for them.  This has resulted in transcribers needing to know if and when a place was part of the British Empire. It also raises the conundrum of whether we take the date of the census or the birth date of the person if a country was not always part of the Empire (e.g. America). This is the first point of our data becoming suspect.
The second issue is that a person born in France, for instance, could be OVB or OVF. OVF is the default. Transcribers might decide that a person was OVB when there is no nationality added to the census page because the parents were born in England, then the child was born in France, and then they all returned to England. Other transcribers would record this as OVF if the census did not say British Subject.
 Then we have places including Canada where for some reason transcribers decide it is OVF not OVB. This should be picked up at validation but validators are forcing it through by putting incorrect entries in the Gazetteer. 

E0 U 6 S69 C8 : P 23
None of this really mattered until we decided to develop FreeCEN2. The Gazetteer was supposed to help sort this out. However validators are now adding entries like France Paris in OVB to the Gazetteer. This is patently incorrect but it speeds up validation. The purpose of validation is data integrity but that point is lost on some people!
We have now reached a point where the distinction between records showing  OVB or OVF is irrelevant. They are effectively both the same.
Now that Kirk is beginning to look at latitude / longitude searches on civil parishes the issue is beginning to flag future problems. Civil Parishes are OK because they are not overseas.
I feel that now is the time to address the issue. We now have a field called Nationality, even in some years when it is not a formal column on the census form. This is to capture the British, German etc Nationalities.
 So, if we stop using the incorrect Chapman Code of OVB and only use OVF and massage our existing data to accommodate that searches on overseas Place of Birth will work properly. Transcribers will no longer have to fathom out whether an entry is OVB or OVF and our data will be much sounder. Coding will be far less messy.
This will require some changes:
I will need to issue an advice that will prepare volunteers for the change.
All entries in the Gazetteer that have a Chapman Code of OVB will need a global replace to OVF.
All existing entries that have an entry of OVB in the database will need to have 'British' added to the Nationality as long as the Nationality is null
All existing entries in the database of OVB will then need to be globally replaced by OVF.

At this point there are a number of disabled places that have been disabled because the Chapman Code is incorrect. The ones having OVB or OVF can be amended to OVF and reinstated to the Gazetteer once the changes have been made.

This will fix all future entries being uploaded using csv files. However vld files need to be tested so that OVB is rejected as an incorrect Chapman Code. I am not sure what testing of vld files happens at the moment.

Geoff

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG there are 118,405 birth chapman_codes of OVB in search records; there are 114,896 birth county records with OVB in freecen_individual ;  There are 113,686 verbatim birth chapman_codes records with OVB in  freecen_individual; there are 9,873 verbatim birth chapman_codes records with OVB in  freecen_csv_entries; there are 439 birth chapman_codes records with OVB in  freecen_csv_entries.
I estimate that the rake task code will take a couple of days to write and check. Running the rake task slowly to avoid service impacts i.e. 1000 records an hr it will take 200 hrs to adjust the database; Likely 10 days.
Do you wish me to commence this now or after the vld download correction.

**Comment by geoffj-FUG** [comment]
Kirk

 

I see the vld issue as more important because it is affecting the integrity of the database.

 

The timing of the elimination of OVB is at our discretion.

 

So, we will tackle the vld issue first.

 

Geoff

**Comment by geoffj-FUG** [comment]
This story will remove a catch 22 from the validation process. If a place name needs correction and has a default of OVF but the person is a British Subject  it cannot be amended to the correct place name because the validation process rejects it as an OVB unless the spreadsheet is downloaded, amended and replaced.
This story is also becoming more important as the Gazetteer is continually having incorrect entries added to it. 
The time has come to make this happen. I suggest that the Gazetteer be updated first and the data can be cleaned afterwards. 
I will need 24 hours to provide notice to Coordinators prior to the changes happening in the Gazetteer. They can then make global changes to any spreadsheets in validation.
Geoff

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG still on hold awaiting confirmation that the vld download issues have been resolved.

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG @PatReynolds 
I am now in a position where I can start to work on this remaining task assigned to me. But before I do can you confirm that you wish to proceed. If so I will convert this task to an Epic and create the stories required for its implementation. Any announcement should likely have multiple parts. The first announcement on what is going to be done and why. This should be followed by other announcements as we proceed with the implementation.

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG 
Before going further I need to explore the implication on VLD uploads and the VLD records with you as all I have seen todate deals with CSVProc. 
As noted earlier 95% of the OVB entries are from vld files. Clearly we must stop OVB entries from new VLD and reloaded files. But how are we to deal with the entries. Nationality does not exist in the freecen_individual collection. I presume we do not want to loose this information. It is quite practical to add a nationality field into the collection and to display it in a similar manner to that proposed for CSVProc.
However I need to know your expectations before proceeding with detailed planning.

**Comment by geoffj-FUG** [comment]
The following reply sent on 13 April does not appear to have reached the Board:
Kirk

Up to now Nationality would have been recorded in the Notes column if it was recorded on the census (though I have seen it added to the POB in brackets).

I assumed that we would just change the OVB to OVF for vld files. 

The Pros - Your suggestion of adding the Nationality to the record set is a better solution. We can therefore add the British Subject info to every OVB record in that set as we do the changes. I see that as a lot tidier as it provides consistency.

Now, the big con – If Bev or Neil load a new vld file it will not have the Nationality field and Valdrev will not accommodate it. My initial thoughts are that they would add it to the Notes column and therefore capture the info that way. That negates the necessity to create the Nationality field for vld files because it cannot be populated in the future. This is just another problem generated by continuing with vld files instead of phasing them out.

I see the con as outweighing the pro. For csv files we should add British Subject to the Nationality field if that field is null. For vld files we should add British subject to the Notes field at the end of any Note that is already there. (It could get messy if there is a note about Nationality there already. But it is better than not adding it at all).

It will be possible for the Coordinator to adjust this as vld files get replaced by the equivalent csv file when we get researcher Action Reports and need to adjust the entry in a vld file. I have replaced a few vld files already. It will take a long time but it will gradually happen.

As an aside I have the updated Transcribers Handbook ready to go for when we introduce the change to OVB / OVF.

Geoff

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG 
My strategy would be to make no change to the vld record structure. Vld files would continue to use OVB as now. The processing would convert to OVF and lnsert British in the Nationality field

**Comment by geoffj-FUG** [comment]
Kirk

 

That sounds like a good idea. It captures the information consistent with the csv files.

 

There will need to be a tweak to vld downloads so that the Nationality field gets filled.

 

Geoff

**Comment by Captainkirkdawson** [comment]
Om hold again until I feel better

**Comment by Captainkirkdawson** [comment]
Given my situation the scrum may wish to reassign

**Comment by geoffj-FUG** [comment]
Anne
This had got to the top of my pile and we were about to start on it when Kirk became too ill to continue.
There are several steps to this:
1. I need to send out an advice that a change is happening. I also have an updated Handbook version ready for issue.
2. About a week later the Gazetteer needs to be changed by removing OVB Overseas British from the County options for Creating New Places.
In conjunction with this change all OVB entries in the Gazetteer need to be amended to OVF.
3. The Csv collection will need to be updated in two fields. If the Chapman Code is OVB and the Nationality field is null, then British subject needs to be added to the Nationality field. The Chapman Code then needs to be changed to OVF.
4. The Vld collection will not need to be updated (see previous discussions)
5. The download of the vld file will need to be amended so that the Nationality field is populated for records that have OVB as a Chapman Code and the OVB Chapman Code downloaded as OVF. (See Kirk's strategy in the previous posts).
6. I believe that the above changes should not affect the presentation of information to researchers.
The elimination of OVB will resolve a number of problems with validation, make transcribing simpler as the transcriber does not need to interpret what they see, and provide a more robust Gazetteer.
Geoff

**Comment by geoffj-FUG** [comment]
Pat
This is a BIG programming task. It needs a routine to be set to crawl through the csv dataset and amend every necessary entry. It has to be done slowly or it will bring down the system.
We can do the changes listed in steps but once started it has to be finished. It will cause long term chaos if we do part of it and then put it on hold.
Maybe this is one for Richard of Sussex or the new programmer to get to know the system with. Although it is big it is relatively straight forward.
Geoff

**Comment by geoffj-FUG** [comment]
Kirk

 

I have attached a draft advice to volunteers for your feedback, please. 

 

I have kept the steps simple reflecting the major milestones of the change. 

 

I have cc’d you as the attachment may not go to Github.

 

Geoff

**Comment by geoffj-FUG** [comment]
Kirk

 

Up to now Nationality would have been recorded in the Notes column if it was recorded on the census (though I have seen it added to the POB in brackets).

 

I assumed that we would just change the OVB to OVF for vld files. 

 

The Pros - Your suggestion of adding the Nationality to the record set is a better solution. We can therefore add the British Subject info to every OVB record in that set as we do the changes. I see that as a lot tidier as it provides consistency.

 

Now, the big con – If Bev or Neil load a new vld file it will not have the Nationality field and Valdrev will not accommodate it. My initial thoughts are that they would add it to the Notes column and therefore capture the info that way. That negates the necessity to create the Nationality field for vld files because it cannot be populated in the future. This is just another problem generated by continuing with vld files instead of phasing them out.

 

I see the con as outweighing the pro. For csv files we should add British Subject to the Nationality field if that field is null. For vld files we should add British subject to the Notes field at the end of any Note that is already there. (It could get messy if there is a note about Nationality there already. But it is better than not adding it at all).

 

It will be possible for the Coordinator to adjust this as vld files get replaced by the equivalent csv file when we get researcher Action Reports and need to adjust the entry in a vld file. I have replaced a few vld files already. It will take a long time but it will gradually happen.

 

As an aside I have the updated Transcribers Handbook ready to go for when we introduce the change to OVB / OVF.

 

Geoff

**Comment by geoffj-FUG** [comment]
Kirk

 

Yes, I would like to proceed please.

 

We got back from seeing the Grandchildren this evening. I will put together a briefing in the next couple of days. (I will try to fit it in tomorrow).

 

Geoff

**Comment by geoffj-FUG** [comment]
The time has come for us to implement this change. VLD uploads have ceased and so the vld data collection is now stable. The steps below outline the changes that need to be made. The key one is the change to the Gazetteer (Step 2). That is where the volunteers will perceive the change. The other changes can happen in the background over the following few weeks.

Scheduling will need to be done by @AnneV-Learn  and Vino. Given the load on the system identified by Kirk above the changes will need to be done when no other major changes having an affect on the system. If a date for the Gazetteer change can be identified in three or four weeks time (say 1 June?) then everything else can follow.  Of course development of any change routines will also have to be taken into account.

This will require some sequential changes:
Step 1 - I will need to issue an advice that will prepare volunteers for the change. (The latest version of the Handbook April 2023 already has the new instructions). This advice will identify the date of the change in the Gazetteer and what action is to be taken by each volunteer.

Step 2 - All entries in the Gazetteer that have a Chapman Code of OVB will need a global replace to OVF. We will need to check for duplicates once that is done. The OVB option will need to be removed from the Chapman Code options for searches and for additions.

Step 3 All existing entries that have an entry of OVB in the vld data collection will need to have ' British' added to the Notes. This will need to be a concatenation to any existing Notes. (Note the space before the B to allow for concatenation). All existing entries in the vld data collection of OVB will then need to be globally replaced by OVF.

Step 4 - All existing entries that have an entry of OVB in the csv data collection will need to have 'British' added to the Nationality as long as the Nationality is null. All existing entries in the csv data collection of OVB will then need to be globally replaced by OVF.

Step 5 – The Download and Convert to Modern Headers process under Show against the file name will need to be amended. During the Conversion the Notes will need to be tested for the string ‘British’. If the string exists, then ‘British’ will need to be entered into the Nationality field and the string removed from the Notes field.

All future transcriptions will only use the OVF Chapman Code and will use the Nationality field when an Nationality entry has been enumerated. Transcribers will no longer be required to interpret a Nationality.

Geoff

**Comment by AnneV-Learn** [comment]
@geoffj-FUG 
**Re - Step 2 of your steps**

A. I have written a rake task that will update Gazetteer entries that have a Chapman Code of OVB to OVF unless there is already a OVF Gazetteer entry for the same standard place name. It runs in 2 modes - fix or just list what the updates would be. 
I have run it on Test3 just to show you what the output looks like (it was run in 'List' mode so no actual updates were made). If an OVF Gazetteer entry already exists for the same place_name it will delete the OVB record, I am trying to avoid duplicates in the Gazetteer as I think it will have an adverse effect of Search results.

B. You say 'The OVB option will need to be removed from the Chapman Code options for searches and for additions.' I am not sure it can be removed from the options for searches until the Steps 4 and 5 have been completed????

Anyway I'll email you the output from the rake mentioned above when run on Test3 - for you to take a look at and assess if this is the way to go.

**Comment by DeniseColbert** [comment]
Aim for 9th June for deployment then running rake

**Comment by AnneV-Learn** [comment]
@geoffj-FUG 
Now that we have rake task for Step 2 scheduled for 9th June I moved on to coding steps 3, 4 and 5.
I have now written the rake task that will cover steps 3 and 4, also the code changes that will take care of step 5.
I tested out a few strategies for the Steps 3 and 4 rake task and found that the most efficient processing approach seemed to be to process each census year at a time. So that is how it will run. It has a parameter to limit the number of records it processes so that we can break it down into chunks to avoid too much load on the system. There is also a parameter which causes the rake task to just list the updates that would occur without actually making the updates. 
There are currently approximately 130,000 search records (with associated VLD/CSV records) in the production data that have Birth County = OVB. 
I did some testing on Test3 in ‘list only mode’ and it handled 10,000 records in 5 mins, to do updates will take longer of course. So I propose that when we run it in production we limit it to 10,000 records per run, with a single run in the early hours of the morning each day. This would spread the whole task over 2 weeks and hopefully avoid any impact to regular users. 
So the next step I think is for you to do some testing on Test3 - I know you are incredibly busy at the moment so don’t expect you to do this anytime soon. 

I propose the following as a testing approach for steps 3 and 4:
I run it on Test3 in ‘list’ only mode, limited to 200 records. 
The output will show what it would be updating if run in ‘fixit’ mode. You could check some of those by doing regular Searches in Test3 to check that it is reporting records that do indeed need updating etc. The listing would give you enough information to do that. 
If all seems ok to you I would then run it again in ‘fixit’ mode (it may not necessarily hit the same 200 records but there is a fair chance it would). Using the output listing from that run you could again do some regular Searches again to check all seems as you would expect after the updates. 
_We have to bear in mind that there are quite a few 'data inconsistencies' in the Test3 data and so some 'Errors' will be pop up in the listing. E.g. Search records where there is no source CSV or VLD data. These are just reported and should not cause the task to crash - I would hope/expect that they will not occur at all when run against the production data._

To test step 5, you will need to load a traditional VLD file with some occurrences of Birth County = OVB and then do  'Convert to Modern Headers' and review the output file.

Does that sound do-able/reasonable?

@Vino-S copying you in incase you have any comments on this approach.

**Comment by geoffj-FUG** [comment]
Anne

 

That looks doable.

 

I am hoping that I will complete my last civil parish correction today. It is a small one.

 

Geoff

**Comment by geoffj-FUG** [comment]
Initial advice to Transcribers and Coordinators issued.
Geoff

**Comment by geoffj-FUG** [comment]
@AnneV-Learn - I have now uploaded a number of SOM vld files to test3. There are also a significant number of SOM csv files already on test3 as well.
Can you run me a report on all OVB and OVF entries in the county of SOM for each of these data sets please. The report should include Piece number, folio, schedule number, surname, POB (Chapman and Place), Alternative POB (Chapman and place). Output should be csv.
Once you have got your programme for updating ready can you run it on SOM in each of the test3 data sets and then send me updated copies of the csv files.
I can then examine the changes to see that all is well. We will keep the test to SOM records so that the other counties are untouched for other testing purposes.
Thanks
Geoff

**Comment by geoffj-FUG** [comment]
See also story #684 ad #688. When this story is completed in June these other stories will no longer be needed.
Geoff

**Comment by AnneV-Learn** [comment]
@geoffj-FUG to run in the most efficient way the program for updating works through census years not counties. However I can add a county restriction just for testing purposes. This means it will ‘touch’ all CSV and VLD data in SOM not just data relating to files you have uploaded. The listing (yet to be written) will therefore cover all VLD and CSV files for SOM. Hope that is ok with you.

**Comment by geoffj-FUG** [comment]
Anne

 

I was just using the SOM files as the test files and preserving the other counties at this time. When we run properly I assumed that it would be the lot, not county by county.

 

My restriction was just so that w we could run a test in test3 and make sure the results were correct.

 

Geoff

**Comment by AnneV-Learn** [comment]
@geoffj-FUG yes of course I understand that.

**Comment by AnneV-Learn** [comment]
@geoffj-FUG another point to note is that the update only touches incorporated CSVProc files. This is done because the most efficient way to tackle the updates is starting at the Search records (these are the records generated from the CSV and VLD data when data is incorporated (CSVProc) or loaded (VLD). Also any ‘unincorporated’ data is in a transient state and so the system could get very inconsistent if the update process is changing data which is still being validated. After 09/06/23 OVB will no longer be in the gazetteer so ongoing validation will take care of the unincorporated CSVProc data. Also update process can continue to run for a few weeks after it has reported that there are no more records to update - it will
just run and check that that is still true and if not update them - this will catch any wayward records that slip through. Ok?

**Comment by geoffj-FUG** [comment]
Anne

 

Thanks for that.

 

I sent the following advice to Coordinators over the weekend:

 

All Coordinators

On Friday 9th June we will be making a major change to the OVB / OVF Chapman Codes. The Gazetteer will be changed so that all overseas entries are OVF. There will be no OVB entries and therefore OVB will no longer be a valid Chapman Code or used in transcriptions.

The amendments to the Gazetteer will take place on Friday 9th June. That will affect any piece that is in validation on that day. You will need to download the piece, amend the OVB entries to OVF and Replace the piece in the system. You can then continue to validate in the normal manner.

We have about 130,000 entries in the data with the Chapman Code of OVB. We have run a small test report and it takes about 5 minutes to report on 10,000 entries, without any other actions taking place on the data. We therefore estimate that it will take about 10 to 15 minutes to amend 10,000 entries.

The changes being made to the data will be:

All existing entries that have an entry of OVB in the FreeCEN1 vld data will have 'British' added to the Notes. This will be a concatenation to any existing Notes. The entry of OVB will be changed to OVF.

All existing entries that have an entry of OVB in the FreeCEN2 csv data will have 'British' added to the Nationality provided that the Nationality is blank. The entry of OVB will be changed to OVF.

We will be running a routine to amend 10,000 entries per day starting after the 9th of June (the start exact date is not yet decided). Only 10,000 entries will be amended each day to limit the load on the system and the effect on users. This routine will be run at about 2:00 am UK time, and so the run will have a minimal effect on UK volunteers or the public. This will be repeated at about 2:00 am each morning for about a fortnight.

The amendments should all have been made to our data by about the end of June. 

The Download and Convert to Modern Headers process used by Coordinators will also be amended to allow for this change. You will be advised when this amendment has been made.

Geoff

**Comment by AnneV-Learn** [comment]
@geoffj-FUG I’ve run a listing that should give you the info you need, I’ll email it to you separately. It has to be said that there is a huge amount of ‘inconsistent’ data in Test3 and the listing found many records where the correct links from the search records to the CSV/VLD entry records did not exist or were wrong. This will have happened as the system was evolving and being tested I guess. Anyway, the listing program reported lots but they have not been included in the actual listing I will send. 
Let me know if you are happy with the listing and I will then run the OVB -> OVF updates across SOM only on Test3.

**Comment by geoffj-FUG** [comment]
Anne

I have had a look at the results of each report and I am more than happy.

All ready to go on 9th June

Geoff

**Comment by geoffj-FUG** [comment]
Note: The Caption for OVF for the Gazetteer and for search results will need to be amended to Born Overseas.
Geoff

**Comment by AnneV-Learn** [comment]
@geoffj-FUG 
Just one more thing to test I think - on Test3 load a traditional VLD file with some occurrences of Birth County = OVB and then do 'Convert to Modern Headers' and review the output file. Or did you already try that?

Also did you want to raise a separate story for the caption change or do it in this one?

**Comment by geoffj-FUG** [comment]
Anne

 

The test file HO107_1943 met that requirement (amongst others)

 

Geoff

**Comment by geoffj-FUG** [comment]
Anne

 

The caption change is part of this one. I realised that it had not been covered.

 

Geoff

**Comment by AnneV-Learn** [comment]
🙂👍🏻

**Comment by AnneV-Learn** [comment]
@geoffj-FUG The 'base' list of Chapman codes and their 'captions' are shared by FreeCEN, FreeREG and FreeBMD, as far as I can see. Is the caption update (Overseas Foreign -> Born Overseas) relevant to them too?
Could all get a bit messy!
@Vino copying you in for your thoughts too.

**Comment by geoffj-FUG** [comment]
Anne

 

Neither FreeBMD nor FreeREG would have overseas entries.

 

FreeBMD is based on the GRO records of BMDs in the British Isles whilst FreeREG is based on the Baptism, Marriage and Deaths ecclesiastical records.

 

I can confirm that it will have no impact with Richard Light if you wish.

 

Geoff

**Comment by AnneV-Learn** [comment]
@geoffj-FUG might be as well just to double check with Richard Light

**Comment by AnneV-Learn** [comment]
Hi Geoff,

Just checking - are we definitely good to go with this caption change?
I am away from Friday onwards (until Wednesday 7th June) and want to get everything lined up for Vino re the 9th’s production deployment.

Bye for now,
Anne

On 18 May 2023, at 10:42, Geoff J ***@***.***> wrote:

﻿

Anne



Neither FreeBMD nor FreeREG would have overseas entries.



FreeBMD is based on the GRO records of BMDs in the British Isles whilst FreeREG is based on the Baptism, Marriage and Deaths ecclesiastical records.



I can confirm that it will have no impact with Richard Light if you wish.



Geoff

**Comment by geoffj-FUG** [comment]
Anne

 

Yes, all good

 

Geoff

**Comment by AnneV-Learn** [comment]
@geoffj-FUG I have made the OVF Caption update on Test3 if you could just test that please.

**Comment by geoffj-FUG** [comment]
Anne

 

Looks good.

 

Geoff

**Comment by AnneV-Learn** [comment]
@geoffj-FUG 👍 great all set for 9th deployment then.

**Comment by AnneV-Learn** [comment]
@geoffj-FUG The overnight Rake task that updates OVB to OVF across the loaded/incorporated VLD and CSVPro production data has been running since 9th June as planned. It did not run for a few days due to clashes with server updates but has now run across all production data. 
It has highlighted a few data issues where there is a disconnect between the search records and the originating data. These issues are for CSVPro data only and occur in the following 5 Pieces:

**SOM 1911 RG14_14124
SOM 1911 RG14_14125
SOM 1911 RG14_14126
KEN 1871 RG10_922
LND 1871 RG10_455**

The CSV files for these pieces need to be deleted (search records will then be deleted as part of that process) and then the files reloaded.
If you could please arrange for that to be done and let me know when completed, I'll then re-check the data.

The overnight Rake will continue to run to pick up any 'stragglers' (I notice we already have one of these in data that was loaded early this morning - WOR 1891 RG12_2350).

**Comment by AnneV-Learn** [comment]
@geoffj-FUG following on from comments above.
If we are going to continue to allow System_Administrators/Data_Managers/Country Coordinators to Upload/Replace New VLD files. I will need to make updates to the VLD file load process so that it does the same OVB -> OVF updates as the Rake. 
At the moment there is not issue because the overnight Rake task is still running and will trap and update any OVB data as a result of a VLD Load/Replace.
Thoughts?

**Comment by geoffj-FUG** [comment]
Anne

 

County Coordinators can no longer download or Replace vld files. They have to be converted to csv and re-validated.

 

Once we have removed OVB from everything OVB at re-validation will cause an error. It will not exist so there is no need to trap it. The effect will be exactly the same as entering any other invalid Chapman Code.

 

The only time a data manager will replace a file is if they have downloaded it as vld and then replaced it after their work. FCTools does not work on 64 bit machines so it is no longer an option to edit them. I think that we can accept that OVB will no longer be a problem once the changes have been made to the data collections.

 

Geoff

**Comment by AnneV-Learn** [comment]
@geoffj-FUG System_Administrators/Data_Managers/Country Coordinators still have access to load VLD data - should we remove that completely now? 
(Could leave the code in the code base but make sure that the process cannot be initiated by anyone).

**Comment by geoffj-FUG** [comment]
Anne

 

The only people who should have the ability to load a vld file are Administrators and Data Managers.

 

County Coordinators should have had the ability removed already.

 

Geoff

**Comment by AnneV-Learn** [comment]
@geoffj-FUG 
Ok I’ll have to change that but if a sys-admin or dm user does load a VLD file and there are any data with POB OVB there will be a problem unless I update the VLD loader to deal with it as part of the final step of the OVB elimination modifications.

**Comment by geoffj-FUG** [comment]
Anne

 

The only uploads by the data manager will be pieces that she has downloaded. The download and upload usually occur on the same day or sometimes one day apart. 

 

They will all have had the OVB correction made.

 

Geoff

**Comment by AnneV-Learn** [comment]
@geoffj-FUG 
I am referring to uploading a **raw** VLD file. The modifications for OVB do not (and cannot) affect the raw file only the data that is created from it and held in the database is changed. I think we need to discuss at the next Scrum as I feel we may be talking at cross purposes here.

**Comment by AnneV-Learn** [comment]
@geoffj-FUG 
The issues identified by the overnight Rake for the following 5 CSVPro data files (where you re-loaded the data) have now been fully resolved.

SOM 1911 RG14_14124
SOM 1911 RG14_14125
SOM 1911 RG14_14126
KEN 1871 RG10_922
LND 1871 RG10_455

Thanks

**Comment by DeniseColbert** [comment]
Done, closing (1538 will supersede)
